crml/cennzx-spot/src/tests.rs

```diff
@@ -834,15 +834,15 @@ fn get_input_price_for_max_reserve_balance() {
 
 /// Calculate input_amount_without_fee using fee rate and input amount and then calculate price
 /// Price = (input_amount_without_fee * output reserve) / (input reserve + input_amount_without_fee)
-// Overflows as the input reserve, output reserve and input amount is at max capacity(u128)
+// Overflows as the input reserve, output reserve and input amount is at max capacity(Balance)
 #[test]
 fn get_input_price_should_fail_with_max_reserve_and_max_amount() {
 	ExtBuilder::default().build().execute_with(|| {
 		assert_err!(
 			CennzXSpot::get_input_price(
-				u128::max_value(),
-				u128::max_value(),
-				u128::max_value(),
+				LowPrecisionUnsigned::max_value(),
+				LowPrecisionUnsigned::max_value(),
+				LowPrecisionUnsigned::max_value(),
 				DefaultFeeRate::get()
 			),
 			"Overflow error"
```

Comment:
@holygits looks like after refining the calculations we can in fact avoid the overflow in this test and return Ok(169886353929574869427545984738775941814) which is probably correct. So I suspect this test itself is not valid. This is the only test that fails at the moment.

Reply:
Another possible scenario that I'm considering is that we have a silent saturation happening since I have used saturating_mul for high precision calculations. The reason for doing that was high precision integers have already got double space, so I assumed u128 numbers are not capable of saturating u256 results! it could then simplify the code to use saturating operations as they have a simpler interface. However, I may need to re-evaluate this assumption. What do you think?

Reply:
Ok, back to my first comment. My assumption that saturating calculations for u128 values in the space of u256 is safe must be correct.

Reply:
LGTM